packages/react-native/src/primitives/Radio/types.ts

```diff
@@ -0,0 +1,39 @@
+import { PressableProps, StyleProp, TextStyle, ViewStyle } from 'react-native';
+
+import { LabelPosition } from '../Label/types';
+
+export type Size = number | 'small' | 'medium' | 'large';
+
+export interface RadioProps<T> extends PressableProps {
+  disabled?: boolean;
+  label?: string;
+  labelPosition?: LabelPosition;
+  labelStyle?: StyleProp<TextStyle>;
+  onChange?: (value?: T) => void;
+  radioContainerStyle?: StyleProp<ViewStyle>;
+  radioDotStyle?: StyleProp<ViewStyle>;
+  size?: Size;
+  selected?: boolean;
+  value: T;
+}
+
+export interface RadioSize {
+  height: number;
+  width: number;
+}
+
+export interface RadioSizes {
+  radioContainerLarge: RadioSize;
+  radioContainerMedium: RadioSize;
+  radioContainerSmall: RadioSize;
+  radioDotLarge: RadioSize;
+  radioDotMedium: RadioSize;
+  radioDotSmall: RadioSize;
+}
+
+export interface RadioStyles extends RadioSizes {
+  container: ViewStyle;
+  disabled: ViewStyle;
+  radioContainer: ViewStyle;
+  radioDot: ViewStyle;
+}
```

Comment:
Thinking we can replace `ViewStyle` with `RadioDimensions` in the return of `getRadioDimensions`:

```suggestion
export type RadioDimensions = Pick<ViewStyle, 'height' | 'width'>;

export interface RadioStyles {
  container: ViewStyle;
  disabled: ViewStyle;
  radioContainer: ViewStyle;
  radioContainerLarge: RadioDimensions;
  radioContainerMedium: RadioDimensions;
  radioContainerSmall: RadioDimensions;
  radioDot: ViewStyle;
  radioDotLarge: RadioDimensions;
  radioDotMedium: RadioDimensions;
  radioDotSmall: RadioDimensions;
}
```